aws_lambda_powertools/event_handler/middlewares/openapi_validation.py

```diff
@@ -410,7 +410,7 @@ def _normalize_multi_query_string_with_param(
     return resolved_query_string
 
 
-def _normalize_multi_header_values_with_param(headers: Dict[str, Any], params: Sequence[ModelField]):
+def _normalize_multi_header_values_with_param(headers: MutableMapping[str, Any], params: Sequence[ModelField]):
```

Comment:
I'm wondering what the benefit is of having `MutableMapping` here instead of Dict. Are there any additional benefits?

Reply:
This is because the `CaseInsensitiveDict` implementation used in the PR for headers is a subclass of `MutableMapping`, but not of `Dict`.